### PR TITLE
MTM-43022 Fix Build on Apple Silicon

### PR DESCRIPTION
**Pull request by ghost**
This fixes the build of cumulocity-examples on Apple Silicon. It does so by two things:

- It fixes the LinuxGenericHardwareDriver.java . Mac OS X has a network interface without a MAC, which produces a NPE.
- The chronicle map implementation used by SNMP uses a very dated version of JNA, which breaks on Apple Silicon. Second, upgrading the JNA dramatically improves SNMP build speed.

### CODE REVIEW

**Review by bbyreddy — DISMISSED** [review]
Thank you, Elias, for fixing the issue. I am fine with the changes.

**Review by atulpandasag — DISMISSED** [review]

**Review by c8y-kh4mell — DISMISSED** [review]

**Review by bggov — DISMISSED** [review]

File: java-agent/generic-linux-driver/src/main/java/c8y/linux/LinuxGenericHardwareDriver.java [hunk 1]
```diff
@@ -114,8 +115,10 @@ private void initializeUsingNetworkInterface() throws SocketException {
 		if (interfaces.hasMoreElements()) {
```
Comment (c8y-kh4mell) [inline java-agent/generic-linux-driver/src/main/java/c8y/linux/LinuxGenericHardwareDriver.java]: `if` -> `while`

File: java-agent/generic-linux-driver/src/main/java/c8y/linux/LinuxGenericHardwareDriver.java [hunk 2]
```diff
@@ -114,8 +115,10 @@ private void initializeUsingNetworkInterface() throws SocketException {
 		if (interfaces.hasMoreElements()) {
 			NetworkInterface networkInterface = interfaces.nextElement();
 			byte[] mac = networkInterface.getHardwareAddress();
-			String address = Hex.encodeHexString(mac).toUpperCase();
-			hardware.setSerialNumber(address);
+			if (Objects.nonNull(mac)) {
+				String address = Hex.encodeHexString(mac).toUpperCase();
+				hardware.setSerialNumber(address);
```
Comment (c8y-kh4mell) [inline java-agent/generic-linux-driver/src/main/java/c8y/linux/LinuxGenericHardwareDriver.java:120]: add `return;` when mac found

**Review by sourajitsaha17 — APPROVED** [review]

**Review by bggov — APPROVED** [review]

**Comment by ghost** [comment]
Pull request is obsolete.

**Pull request closed without merging**